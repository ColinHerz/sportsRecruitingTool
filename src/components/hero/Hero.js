import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import React from "react";

import "./hero.scss";

const Hero = props => {
	if (props.loggedIn) {
		return (
			<main id="hero-logged-in">
				<h2>Welcome to Sporta</h2>

				<p>You can view...</p>

				<ul>
					<li><Link to="/myevents/garbage/">Events</Link></li>
					<li><Link to="/scores/">Top Scores</Link></li>
					<li><Link to="/profile/garbage/">Your Profile</Link></li>
				</ul>
			</main>
		);
	}

	return (
		<main id="hero-logged-out">
			<p>Leverage agile frameworks to provide a robust synopsis for high level overviews. Iterative approaches to corporate strategy.</p>

			<button
				className="sign-up"
				onClick={props.registering}
			>
				Sign Up
			</button>

			<button
				className="log-in"
				onClick={props.loggingIn}
			>
				Log In
			</button>
		</main>
	);
};

Hero.propTypes = {
	loggingIn: PropTypes.func.isRequired,
	registering: PropTypes.func.isRequired,
	loggedIn: PropTypes.bool.isRequired
};

export default Hero;
