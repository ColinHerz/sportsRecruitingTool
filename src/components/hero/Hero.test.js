/* eslint-env jest */

// MemoryRouter needed because header contains NavLinks, which need to
// be rendered within a Router
import { MemoryRouter } from "react-router-dom";
import React from "react";
import renderer from "react-test-renderer";

import Hero from "./Hero.js";

const genProps = loggedIn => {
	return {
		logIn: jest.fn(),
		register: jest.fn(),
		loggedIn: loggedIn
	};
};

describe(`Hero component`, () => {
	beforeEach(() => {
		jest.resetModules();
	});

	it(`renders correctly when not logged in`, () => {
		const props = genProps(false);

		const component = renderer.create(
			<MemoryRouter>
				<Hero {... props} />
			</MemoryRouter>
		);
		const tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it(`renders correctly when logged in`, () => {
		const props = genProps(true);

		const component = renderer.create(
			<MemoryRouter>
				<Hero {... props} />
			</MemoryRouter>
		);
		const tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
