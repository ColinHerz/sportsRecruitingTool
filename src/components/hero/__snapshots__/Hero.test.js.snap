// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero component renders correctly when logged in 1`] = `
<main
  id="hero-logged-in"
>
  <h2>
    Welcome to Sporta
  </h2>
  <p>
    You can view...
  </p>
  <ul>
    <li>
      <a
        href="/myevents/garbage/"
        onClick={[Function]}
      >
        Events
      </a>
    </li>
    <li>
      <a
        href="/scores/"
        onClick={[Function]}
      >
        Top Scores
      </a>
    </li>
    <li>
      <a
        href="/profile/garbage/"
        onClick={[Function]}
      >
        Your Profile
      </a>
    </li>
  </ul>
</main>
`;

exports[`Hero component renders correctly when not logged in 1`] = `
<main
  id="hero-logged-out"
>
  <p>
    Leverage agile frameworks to provide a robust synopsis for high level overviews. Iterative approaches to corporate strategy.
  </p>
  <button
    className="sign-up"
    onClick={[MockFunction]}
  >
    Sign Up
  </button>
  <button
    className="log-in"
    onClick={[MockFunction]}
  >
    Log In
  </button>
</main>
`;
